components/dashboard/src/projects/ConfigureProject.tsx

```diff
@@ -166,23 +164,25 @@ export default function () {
         </Suspense>
         {isDetecting && <div className="absolute h-full w-full bg-gray-100 dark:bg-gray-700 flex items-center justify-center space-x-2">
           <img className="h-5 w-5 animate-spin" src={isDark ? SpinnerDark : Spinner} />
-          <span className="font-semibold text-gray-400">Detecting project type ...</span>
+          <span className="font-semibold text-gray-400">Detecting project configuration ...</span>
```

Comment:
**question**: This is taking noticeably more time sometimes, like 4-5 seconds or more when there's no git based configuration. Expected?

Reply:
Yes, I think we should look if we can do that ahead of time.

Reply:
We could also probably optimize the "config inferrer": It currently makes repository requests one by one (e.g. "does a `yarn.lock` file exist? Oh... does a `cargo.toml` file exist? Oh... etc.)

Fetching the entire repo structure (not necessarily file content or history) in one go, and using that to respond to the "config inferrer"'s queries would already be a huge performance boost.

Reply:
Now tracked as https://github.com/gitpod-io/gitpod/issues/5527